src/worker.ts

```diff
@@ -1,11 +1,39 @@
 import "unfetch/polyfill";
 import { CONFIG_URL } from "./constants";
+import Fetch from "./tasks/fetch";
+import Pop from "./tasks/pop";
+import Task from "./tasks/task";
 
-interface Response {
-  json(): Promise<Config>;
+interface TaskLookup {
+  [key: string]: typeof Task;
 }
 
-export function init(): Promise<Config> {
-  return fetch(CONFIG_URL).then((res: Response): Promise<Config> => res.json());
+const TASK_LOOKUP: TaskLookup = {
+  pop: Pop,
+  fetch: Fetch
+};
+
+function runTasks(configuration: Config): Promise<Beacon[]> {
+  const promises = configuration.tasks.map(
+    (taskData): Promise<Beacon> => {
+      const TaskClass = TASK_LOOKUP[taskData.type];
+      if (TaskClass) {
+        const task = new TaskClass(configuration, taskData);
+        return task.execute();
+      }
+      return Promise.reject(`Unknown task type: ${taskData.type}`);
+    }
+  );
+  return Promise.all(promises);
```

Comment:
This fine for now/testing, however we want to process the task queue sequentially not in parallel. See this util [`src/util/squence.js`](https://github.com/fastly/insights.js/blob/master/src/util/sequence.js) from v1.